config/configuration_service.proto

```diff
@@ -7,65 +7,128 @@ package config;
 
 service ConfigurationService {
     //Used when you just need to read the current configuration
-    rpc GetConfiguration(ReadRequest) returns (ConfigurationResponse) {}
+    rpc GetConfiguration (ReadRequest) returns (ConfigurationResponse) {
+    }
 
     //Used when you want to update the existing configuration
-    rpc UpdateConfiguration(UpdateRequest) returns (ConfigurationResponse) {}
+    rpc UpdateConfiguration (UpdateRequest) returns (ConfigurationResponse) {
+    }
 
     //This is used to instruct the Daemon to stop taking in any new requests,
     //At this point , however any existing requests being processed will be honored.
     // ("_Request_Stop", "block_number",authentication_address) should be sent in the signature
-    rpc StopProcessingRequests(CommandRequest) returns (Response) {}
+    rpc StopProcessingRequests (CommandRequest) returns (Response) {
+    }
 
     //Restores the normal behavior to start processing new requests again
     // ("_Request_Start", "block_number",authentication_address) should be sent in the signature
-    rpc StartProcessingRequests(CommandRequest) returns (Response) {}
-}
+    rpc StartProcessingRequests (CommandRequest) returns (Response) {
+    }
+
+    //To Determine if the Daemon is processing requests are not.
+    // ("__Is_Daemon_Processing_Requests", "block_number",authentication_address) should be sent in the signature
+    rpc IsDaemonProcessingRequests (ReadRequest) returns (Response) {
+    }
+
 
 
+}
+
 //Used when you just need to read the current configuration
 message ReadRequest {
     //Signature will compromise of the below
     // ("_Request_Read", "block_number",authentication_address)
+    //OR
+    //("__Is_Daemon_Processing_Requests", "block_number",authentication_address)
     bytes signature = 1;
 
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
```

Comment:
I would propose adding caller authentication message and reusing it in all commands:
```proto
//Caller authentication data
message CallerAuth {
    //Signature will compromise the following data:
    // (method name, current block, user address)
    //Example: ("GetConfigurationScheme", <current_block>) signed by private key
    bytes signature = 1;
    //current block number (signature will be valid only for short time around this block number)
    uint64 current_block = 2;
    //The address of the person who has rights to make configuration changes on Daemon
    string user_address = 3;
}
```

Reply:
I like this , will do